File: Interview Preparation Kit/Dynamic Programming/Candies.java
// https://www.hackerrank.com/challenges/candies/problem
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the candies function below.
    static long candies(int n, int[] arr) {
        int[] up = new int[n];
        int[] down = new int[n];
        up[0] = 1;
        down[n - 1] = 1;
        long result = 0;

        for (int i = 1; i < n; i++) {
            if (arr[i] > arr[i - 1]) {
                up[i] = up[i - 1] + 1;
            }
            else {
                up[i] = 1;
            }
        }
        for (int j = n - 2; j >= 0; j--) {
            if (arr[j] > arr[j + 1]) {
                down[j] = down[j + 1] + 1;
            }
            else {
                down[j] = 1;
            }
        }
        for (int k = 0; k < n; k++) {
            result += (long) Math.max(up[k], down[k]);
        }

        return result;
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            int arrItem = scanner.nextInt();
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
            arr[i] = arrItem;
        }

        long result = candies(n, arr);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}

File: Interview Preparation Kit/Dynamic Programming/Candies.py
#!/bin/python3
# https://www.hackerrank.com/challenges/candies/problem
import math
import os
import random
import re
import sys

# Complete the candies function below.
def candies(n, arr):
    up = [0] * n
    down = [0] * n
    up[0] = 1
    down[n - 1] = 1
    result = 0

    for i in range(1, n):
        if arr[i] > arr[i - 1]:
            up[i] = up[i - 1] + 1
        else:
            up[i] = 1
    for j in range(n - 2, -1, -1):
        if arr[j] > arr[j + 1]:
            down[j] = down[j + 1] + 1
        else:
            down[j] = 1
    for k in range(n):
        result += max(up[k], down[k]);

    return result


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    arr = []

    for _ in range(n):
        arr_item = int(input())
        arr.append(arr_item)

    result = candies(n, arr)

    fptr.write(str(result) + '\n')

    fptr.close()
